paimon-core/src/main/java/org/apache/paimon/partition/PartitionTimeExtractor.java

```diff
@@ -112,7 +112,8 @@ private static LocalDateTime toLocalDateTime(
         DateTimeFormatter dateTimeFormatter =
                 DateTimeFormatter.ofPattern(Objects.requireNonNull(formatterPattern), Locale.ROOT);
         try {
-            return LocalDateTime.parse(timestampString, Objects.requireNonNull(dateTimeFormatter));
+            return LocalDate.parse(timestampString, Objects.requireNonNull(dateTimeFormatter))
+                    .atStartOfDay();
```

Comment:
What about hour partition?

Reply:
If timestampString is "2025051008" and formatterPattern is "yyyyMMddHH".
The exception will not happen.